File: csharp/Validators/ArgumentExistenceValidator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Validators
{
    public class ArgumentExistenceValidator : AbstractValidator
    {
        private ushort Index { get; set; }

        public ArgumentExistenceValidator(List<string> args, ushort index, string message) : base(args, message)
        {
            Index = index;
        }

        public override bool Validate()
        {
            if (Arguments.Count > Index && Arguments[Index] != null)
                return true;

            Console.WriteLine(Message);
            return false;
        }
    }
}

File: java/JavaFileutil/src/JavaFileutil/Validators/ArgumentExistenceValidator.java
package JavaFileutil.Validators;

import java.util.List;

public class ArgumentExistenceValidator extends AbstractValidator
{
    private short Index;

    public ArgumentExistenceValidator(List<String> args, short index, String message)
    {
        super(args, message);
        Index = index;
    }

    @Override
    public boolean Validate()
    {
        if (Arguments.size() <= Index || Arguments.get(Index) == null)
        {
            System.out.println(ErrorPreface + Message);
            return false;
        }

        return true;
    }
}
